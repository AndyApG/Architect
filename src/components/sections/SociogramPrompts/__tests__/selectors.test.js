/* eslint-env jest */

import mockState from '../../../../__tests__/testState.json';

import {
  getLayoutVariablesForNodeType,
  getHighlightVariablesForNodeType,
  getEdgesForNodeType,
} from '../selectors';

jest.mock('redux-form', () => ({
  formValueSelector: () =>
    () => '1234-1234-4',
}));

const nodeType = '1234-1234-1234';
const form = 'edit-prompt';

describe('SociogramPrompts', () => {
  describe('selectors', () => {
    it('get layout variables for node type', () => {
      const result = getLayoutVariablesForNodeType(mockState, { nodeType });

      expect(result).toMatchSnapshot();
    });

    it('get highlight variables for node type', () => {
      const result = getHighlightVariablesForNodeType(mockState, {
        form,
        nodeType,
        formUsedVariableIndex: ['1234-1234-3'],
      });

      expect(result).toMatchSnapshot();
    });

    it('get edges for node type', () => {
      const result = getEdgesForNodeType(mockState, { nodeType });

      expect(result).toMatchSnapshot();
    });
  });
});
