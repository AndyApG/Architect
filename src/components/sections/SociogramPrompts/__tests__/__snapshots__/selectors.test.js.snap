// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SociogramPrompts selectors get edges for node type 1`] = `
Array [
  Object {
    "color": "blue",
    "label": "an edge",
    "value": "1234-5",
  },
]
`;

exports[`SociogramPrompts selectors get highlight variables for node type 1`] = `
Array [
  Object {
    "label": "my boolean",
    "type": "boolean",
    "value": "1234-1234-3",
  },
  Object {
    "label": "other boolean 2",
    "type": "boolean",
    "value": "1234-1234-4",
  },
]
`;

exports[`SociogramPrompts selectors get layout variables for node type 1`] = `
Array [
  Object {
    "label": "my layout",
    "type": "layout",
    "value": "1234-1234-1",
  },
]
`;
